sdk/nodejs/runtime/resource.ts

```diff
@@ -425,8 +472,11 @@ export function registerResource(res: Resource, parent: Resource | undefined, t:
  * properties.
  */
 async function prepareResource(label: string, res: Resource, parent: Resource | undefined, custom: boolean, remote: boolean,
-                               props: Inputs, opts: ResourceOptions): Promise<ResourceResolverOperation> {
-
+    props: Inputs, opts: ResourceOptions, type: string, name: string): Promise<ResourceResolverOperation>;
+async function prepareResource(label: string, res: Resource, parent: Resource | undefined, custom: boolean, remote: boolean,
+    props: Inputs, opts: ResourceOptions): Promise<ResourceResolverOperation>;
+async function prepareResource(label: string, res: Resource, parent: Resource | undefined, custom: boolean, remote: boolean,
+                               props: Inputs, opts: ResourceOptions, type?: string, name?: string): Promise<ResourceResolverOperation> {
```

Comment:
Do we even need to bother with the overloads? Wondering if we just keep the single definition with `type?: string, name?: string` added.

If we do keep the overloads, (nit) can we indent the `props: ...` line for each consistently?

Reply:
I'm fine with the single definition and optional args - I can definitely pull that back (I wanted to keep the original signature here just so consumers of the current `prepareResource` function won't see changes they don't expect if they upgrade their SDK).